### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by Maxb0tbeep** [ntdevlabs/tiny11builder#48]
can I run this on linux?

**Comment by ianis58** [comment]
all this script is windows specific you have no chance to get it to run on linux.

**Comment by InsertX2k** [comment]
Short answer no.

Long answer, You can't run this on Linux because dism and reg utilities aren't available under a Linux environment.

**Comment by s3rj1k** [comment]
> Long answer, You can't run this on Linux because dism and reg utilities aren't available under a Linux environment.

Long answer is yes, use WINE

**Comment by ianis58** [comment]
I really think that the "dism" utility will not work / miss dependencies if you try to run it on wine.
You can't run it on Linux except if you do in a windows virtual machine.
You need to run it on an up to date windows 10.

**Comment by s3rj1k** [comment]
> I really think that the "dism" utility will not work / miss dependencies if you try to run it on wine. You can't run it on Linux except if you do in a windows virtual machine. You need to run it on an up to date windows 10.

https://wimlib.net/

Worked fine enough when I was using it in Linux for Windows 8 ISO

**Comment by ianis58** [comment]
Impressive I did not knew that something like that exists!
Still it would need to rewrite the whole script to use wimlib instead of dism, but yeah definitely possible with that!

**Comment by s3rj1k** [comment]
> Impressive I did not knew that something like that exists! Still it would need to rewrite the whole script to use wimlib instead of dism, but yeah definitely possible with that!

I am not sure that it will work with `/Remove` commands for component removal but is does work fine with imaging part.

So, you can defiantly mount wim under Linux and from there abuse WINE prefixes to use mounted Windows image DLLs and use dism binary from the actual image.

But I would still recommend using VMs :)

**Comment by InsertX2k** [comment]
> > Long answer, You can't run this on Linux because dism and reg utilities aren't available under a Linux environment.
> 
> Long answer is yes, use WINE

Although `reg.exe` is bundled within the binaries of **Wine**, `Dism.exe` isn't.

**Comment by jp06** [comment]
@InsertX2k yeah, but is it not possible to use the `DISM.exe` from the image like he said?

**Comment by draylegend** [comment]
@s3rj1k did you use it in wine?

**Comment by s3rj1k** [comment]
> @s3rj1k did you use it in wine?

Yep I used wine :)

**Comment by InsertX2k** [comment]
> @InsertX2k yeah, but is it not possible to use the `DISM.exe` from the image like he said?

`dism.exe` uses the Windows's Virtual Disk service, which isn't available or can't be used on Wine in Linux

**Comment by s3rj1k** [comment]
> > @InsertX2k yeah, but is it not possible to use the `DISM.exe` from the image like he said?
> 
> `dism.exe` uses the Windows's Virtual Disk service, which isn't available or can't be used on Wine in Linux

https://bugs.winehq.org/show_bug.cgi?id=41377

**Comment by InsertX2k** [comment]
> > > @InsertX2k yeah, but is it not possible to use the `DISM.exe` from the image like he said?
> > 
> > 
> > `dism.exe` uses the Windows's Virtual Disk service, which isn't available or can't be used on Wine in Linux
> 
> https://bugs.winehq.org/show_bug.cgi?id=41377

Still, if the VDS service works properly there might be other services required by `dism.exe` that can't be started on linux.

**Comment by jp06** [comment]
@s3rj1k a bit off-topic but do you think it's possible to have IIS on Wine?

**Comment by s3rj1k** [comment]
> @s3rj1k a bit off-topic but do you think it's possible to have IIS on Wine?

Probably not in production grade working state and breaking every possible licenses.

WinPE would be my second choice.
